src/patches.rs

```diff
@@ -89,25 +95,165 @@ fn collect_pickup_resources<'r>(gc_disc: &structs::GcDisc<'r>, starting_items: &
             _ => panic!(),
         };
 
+        // Iterate through all resources in pak //
         for res in pak.resources.iter() {
+            // If this resource is a dependency needed by the patcher, add the resource to the output list //
             let key = (res.file_id, res.fourcc());
             if looking_for.remove(&key) {
                 assert!(found.insert(key, res.into_owned()).is_none());
             }
         }
     }
 
+    // Remove extra assets from dependency search since they won't appear     //
+    // in any pak. Instead add them to the output resource pool. These assets //
+    // are provided as external files checked into the repository.            //
     for res in crate::custom_assets::custom_assets(&found, starting_items) {
         let key = (res.file_id, res.fourcc());
         looking_for.remove(&key);
         assert!(found.insert(key, res).is_none());
     }
 
+    if !looking_for.is_empty()
+    {
+        println!("error - still looking for {:?}", looking_for);
+    }
+
+    assert!(looking_for.is_empty());
+
+    found
+}
+
+
+// Door assets are not shared across all areas either,
+// so we have to make a cache for them as well.
+fn collect_door_resources<'r>(gc_disc: &structs::GcDisc<'r>)
+    -> HashMap<(u32, FourCC), structs::Resource<'r>>
+{   
+    // Get list of all dependencies needed by custom doors //
+    
+    let mut looking_for = HashSet::<_>::new();
+
+    {
+        let looking_for_door: HashSet<_> = DoorType::iter()
```

Comment:
Can this block be written as `looking_for.extend(DoorType::iter().flat_map(...))` and `looking_for.extend(BlastShieldType::iter().flat_map(...))`?

Reply:
I'm not sure how you could specify to map just the dependencies here, and not all of the type enums